<!-- Aggregated PC Ground Plane Extraction Node
  
Launches a ROS node to aggregate incoming sensor_msgs::PointCloud2
over a configurable time horizon. Each input cloud also triggers a
ground plane extraction / publication.
-->
<launch>
  <!-- required args -->
  <arg name="cloud_topic" doc="Input cloud topic to process." />
  <arg name="fixed_frame" doc="Fixed frame in which to assemble clouds." />

  <!-- optional args -->
  <arg name="map_topic" default="map" doc="Map topic to publish out." />
  <arg name="horizon" default="10.0" doc="Time horizon to limit aggregation." />
  <arg name="voxel_size" default="0.5" doc="Voxel size - this implicitly maps to output map cell size." />

  <!-- launch core node -->
  <node name="receding_ground_extraction" pkg="openvdb_voxel_mapping" type="receding_ground_extraction_node" output="screen" required="true">
    <param name="cloud_topic" value="$(arg cloud_topic)" />
    <param name="fixed_frame" value="$(arg fixed_frame)" />
    <param name="map_topic" value="$(arg map_topic)" />
    <param name="horizon" value="$(arg horizon)" />
    <param name="voxel_size" value="$(arg voxel_size)" />
  </node>

</launch>
